libraries/eosiolib/eosiolib.cpp

```diff
@@ -65,6 +69,48 @@ namespace eosio {
       ds >> params;
    }
 
+   //  considering of runtime efficiency, using a estimated size to save the exact size calculating time
+   const int estimate_param_ids_buff_size = 128;
+   const int estimate_params_buff_size = 256;
```

Comment:
Is it possible to add EOS_ASSERT() to make sure ids_buff and buff always less than these consts?

Reply:
Good idea, I add eosio::check to check that, in this file otherwhere all use eosio::check.